### PR TITLE
[ENG-4591] Barium/upgrade ember 3.28

### PR DESCRIPTION
**Pull request by brianjgeiger**
-   Ticket: [ENG-4591]
-   Feature flag: n/a

## Purpose

Upgrade ember to 3.28

## Summary of Changes

1. Match ember dependencies to `ember new` blueprint
2. Fix all the problems
3. Modify getApiErrorMessage to re-throw in the case of no API error message

## QA Notes

Hopefully between tests and selenium we should be covered.

[ENG-4591]: https://openscience.atlassian.net/browse/ENG-4591?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by futa-ikeda — APPROVED** [review]
I can't shake the feeling that there's something not quite right with our files relationships, but I think it's the most-correct it's been. Just one small question about one of our tests, but everything looks good!

File: tests/integration/components/review-actions-list/review-action/component-registration-test.ts [hunk 1]
```diff
@@ -225,25 +225,4 @@ module('Integration | Component | Registration | review-actions', hooks => {
             .doesNotContainText('moderator', 'Review action does not mention moderator');
         assert.dom('[data-test-review-action-comment]').doesNotExist('Empty strings do not create comment');
     });
-
-    test('Request embargo termination action', async function(this: ThisTestContext, assert) {
```
[the remark this answers is not in the document]
Comment (brianjgeiger) [reply tests/integration/components/review-actions-list/review-action/component-registration-test.ts:229]: It was a duplicate test. The updated linter picked it up and said we had two tests with the same name, and I looked through and they seemed to be the same.

File: lib/osf-components/addon/components/contributors/manager/component.ts [hunk 2]
```diff
@@ -114,21 +114,27 @@ export default class ContributorsManager extends Component {
         }
     }
 
+    @action
```
Comment (adlius) [inline lib/osf-components/addon/components/contributors/manager/component.ts:117]: Why do we have to wrap this in an action?

**Review by adlius — APPROVED** [review]
Just one question.

File: lib/osf-components/addon/components/contributors/manager/component.ts (hunk 2 above)
Comment (brianjgeiger) [reply lib/osf-components/addon/components/contributors/manager/component.ts:117]: This one was @futa-ikeda's so I'll let him answer it.

Comment (futa-ikeda) [reply lib/osf-components/addon/components/contributors/manager/component.ts:117]: I think the `@action` decorator here is warranted, but the `(action)` helper in the template file definitely isn't needed. I think I got carried away seeing all the `(perform)s` in there

**Pull request merged**